runtime/bcutil/cfreader.c

```diff
@@ -1783,11 +1798,15 @@ checkMethods(J9PortLibrary* portLib, J9CfrClassFile* classfile, U_8* segment, U_
 		} 
 
 		if (nameIndexOK && utf8Equal(&classfile->constantPool[method->nameIndex], "<init>", 6)) {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES) && defined(J9VM_OPT_NEW_FACTORY_METHOD)
+			if (J9_IS_CLASSFILE_VALUETYPE(classfile)) {
```

Comment:
I see `J9_IS_CLASSFILE_VALUETYPE(classfile)` is used mutiple times in this function, you can cache that in a local `BOOLEAN` variable.